core/src/test/java/io/github/tcdl/ResponderTest.java

```diff
@@ -0,0 +1,94 @@
+package io.github.tcdl;
+
+import io.github.tcdl.config.MsbMessageOptions;
+import io.github.tcdl.messages.Message;
+import io.github.tcdl.messages.payload.Payload;
+import io.github.tcdl.support.TestUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.exceptions.verification.TooLittleActualInvocations;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.mockito.Mockito.*;
+
+/**
+ * Created by anstr on 5/26/2015.
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(Responder.class)
+public class ResponderTest {
+    private static MsbMessageOptions config;
+    private static Responder responder;
+    private static Message message;
+    private static String topic;
+
+    @BeforeClass
+    public static void setUp() {
+        config = TestUtils.createSimpleConfig();
+        topic = config.getNamespace();
+        message = TestUtils.createMsbRequestMessageWithPayloadAndTopicTo(topic);
+        responder = new Responder(config, message);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testResponderNullConfigsNullMessageThrowsException() {
+        new Responder(null, null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testResponderNullConfigsThrowsException() {
+        new Responder(null, message);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testResponderNullMessageThrowsException() {
+        new Responder(config, null);
+    }
+
+    @Test
+    public void testResponderNotNullOk() {
+        String topic = config.getNamespace();
+        new Responder(config, message);
+    }
+
+    @Test
+    public void testResponderSendAckOk() throws Exception {
```

Comment:
What's the purpose of this test? I doubt that we need to test whether some private method of the SAME class is invoked. Also this approach is very fragile in case of  refactorings.

I assume that we need to check here that proper message is published via produce mock. Do we have such test?